test/Datadog.Trace.Tests/Util/ArrayHelperTests.cs

```diff
@@ -0,0 +1,101 @@
+using System.Text;
+using Datadog.Trace.Util;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Util
+{
+    public class ArrayHelperTests
+    {
+        private const string TestMessage = "this is a test string";
+
+        [Fact]
+        public void ArrayHelper_Empty()
+        {
+            byte[] empty1 = ArrayHelper.Empty<byte>();
+            byte[] empty2 = ArrayHelper.Empty<byte>();
+
+            Assert.Same(empty1, empty2);
+        }
+
+        [Fact]
+        public void ArrayHelper_Copy_SrcArray()
+        {
+            byte[] source = Encoding.UTF8.GetBytes(TestMessage);
+            byte[] target = ArrayHelper.Copy(source);
+
+            string message = Encoding.UTF8.GetString(target);
+
+            Assert.Equal(TestMessage, message);
+            Assert.NotSame(source, target);
+        }
+
+        [Fact]
+        public void ArrayHelper_Copy_SrcArrayToGivenTarget()
+        {
+            byte[] source = Encoding.UTF8.GetBytes(TestMessage);
+            byte[] target = new byte[source.Length];
+
+            ArrayHelper.Copy(source, target);
+
+            string message = Encoding.UTF8.GetString(target);
+
+            Assert.Equal(TestMessage, message);
+            Assert.NotSame(source, target);
+        }
+
+        [Fact]
+        public void ArrayHelper_Copy_SrcArrayToGivenTargetAndLength()
+        {
+            string sourceMessageAddition = ". Additions.";
+            string sourceMessage = TestMessage + sourceMessageAddition;
+
+            int additionLength = Encoding.UTF8.GetBytes(sourceMessageAddition).Length;
+
+            byte[] source = Encoding.UTF8.GetBytes(sourceMessage);
+            byte[] target = new byte[source.Length - additionLength];
+
+            ArrayHelper.Copy(source, target, source.Length - additionLength);
+
+            string message = Encoding.UTF8.GetString(target);
+
+            Assert.Equal(TestMessage, message);
+            Assert.NotSame(source, target);
+        }
+
+        [Fact]
+        public void ArrayHelper_Copy_SrcArrayToGivenTargetOffsetLength()
+        {
+            string sourceMessagePrefix = "Prefix. ";
+            string sourceMessageSuffix = ". Suffix.";
+            string sourceMessage = string.Concat(sourceMessagePrefix, TestMessage, sourceMessageSuffix);
+
+            int prefixLength = Encoding.UTF8.GetBytes(sourceMessagePrefix).Length;
+            int suffixLength = Encoding.UTF8.GetBytes(sourceMessageSuffix).Length;
+
+            byte[] source = Encoding.UTF8.GetBytes(sourceMessage);
+            byte[] target = new byte[source.Length - prefixLength - suffixLength];
+
+            ArrayHelper.Copy(source, target, prefixLength, 0, source.Length - prefixLength - suffixLength);
+
+            string message = Encoding.UTF8.GetString(target);
+
+            Assert.Equal(TestMessage, message);
+            Assert.NotSame(source, target);
+        }
+
+        [Fact]
+        public void ArrayHelper_Copy_OtherThanPrimitive()
```

Comment:
It would also be good to test the case where the `newSize` and `length` argument do not match

Reply:
Agree. Added 2 more tests.